### PR TITLE
feat: asset selection, quote updating etc

### PR DESCRIPTION
**Pull request by rcresnik**

### CODE REVIEW

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapInteractor.swift [hunk 1]
```diff
@@ -9,32 +9,293 @@
 //
 
 import UIKit
+import WalletKit
 
-class SwapInteractor: NSObject, Interactor, SwapViewActions {
+class SwapInteractor: NSObject, Interactor, SwapViewActions, Subscriber {
```
Comment (kenan-mamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapInteractor.swift]: i don't think we need Subscriber anymore

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapInteractor.swift [hunk 2]
```diff
@@ -9,32 +9,293 @@
...
+        }
+        
+        presenter?.presentChoseCurency(actionResponse: .init(from: from, to: to))
+    }
     
-    func assetSelected(viewAction: SwapModels.SelectedAsset.ViewAction) {}
+    // MARK: - Aditional helpers
+}
+
+extension RangeReplaceableCollection where Self: StringProtocol {
```
Comment (kenan-mamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapInteractor.swift]: can be moved to extensions

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapModels.swift [hunk 3]
```diff
@@ -11,40 +11,88 @@
...
     
     struct SelectedAsset {
         struct ViewAction {
-            var name: String?
+            var from: String?
+            var to: String?
+        }
+    }
+    
+    struct ChoseCuurency {
```
Comment (kenan-mamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapModels.swift]: Choose or Chosen would be better

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapMainVIP.swift [hunk 4]
```diff
@@ -16,27 +16,51 @@ extension Scenes {
...
 }
 
 protocol SwapActionResponses: BaseActionResponses, FetchActionResponses {
     func presentSetAmount(actionResponse: SwapModels.Amounts.ActionResponse)
+    func presentChoseCurency(actionResponse: SwapModels.ChoseCuurency.ActionResponse)
 }
 
 protocol SwapResponseDisplays: AnyObject, BaseResponseDisplays, FetchResponseDisplays {
     func displaySetAmount(responseDisplay: SwapModels.Amounts.ResponseDisplay)
+    func displayChoseCurency(responseDisplay: SwapModels.ChoseCuurency.ResponseDisplay)
```
Comment (kenan-mamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapMainVIP.swift]: Chosen

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapMainVIP.swift [hunk 5]
```diff
@@ -16,27 +16,51 @@ extension Scenes {
 
 protocol SwapViewActions: BaseViewActions, FetchViewActions {
     func setAmount(viewAction: SwapModels.Amounts.ViewAction)
+    func switchPlaces(viewAction: SwapModels.SwitchPlaces.ViewAction)
+    func choseCurency(viewAction: SwapModels.ChoseCuurency.ViewAction)
 }
 
 protocol SwapActionResponses: BaseActionResponses, FetchActionResponses {
     func presentSetAmount(actionResponse: SwapModels.Amounts.ActionResponse)
+    func presentChoseCurency(actionResponse: SwapModels.ChoseCuurency.ActionResponse)
```
Comment (kenan-mamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapMainVIP.swift]: Chosen

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapMainVIP.swift [hunk 6]
```diff
@@ -16,27 +16,51 @@ extension Scenes {
 
 protocol SwapViewActions: BaseViewActions, FetchViewActions {
     func setAmount(viewAction: SwapModels.Amounts.ViewAction)
+    func switchPlaces(viewAction: SwapModels.SwitchPlaces.ViewAction)
+    func choseCurency(viewAction: SwapModels.ChoseCuurency.ViewAction)
```
Comment (kenan-mamedoff) [inline brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapMainVIP.swift]: Choose

File: brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapInteractor.swift (hunk 1 above)
Comment (rcresnik) [reply brd-ios/breadwallet/src/Redesign/Scenes/Swap/Swap/SwapInteractor.swift]: bwo.. ur code :D

**Review by kenan-mamedoff — APPROVED** [review]

**Review by kenan-mamedoff — APPROVED** [review]

**Pull request merged**